python_ta/reporters/node_printers.py

```diff
@@ -153,6 +205,328 @@ def render_blank_line(line):
+    operators = {">>", "<<"}
+    end_idx = col + 1
+    end_idx = end_idx + 1 if source_lines[line][col : col + 2] in operators else end_idx
+
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (msg.line, slice(col, end_idx), LineType.ERROR, source_lines[line])
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e228(msg, _node, source_lines=None):
+    """Render a PEP8 missing whitespace around modulo operator message."""
+    line = msg.line - 1
+    res = re.search(r"column (\d+)", msg.msg)
+    col = int(res.group().split()[-1])
+
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (
+        msg.line,
+        slice(col, col + 1),
+        LineType.ERROR,
+        source_lines[line] + "  # INSERT A SPACE BEFORE & AFTER THE MODULO OPERATOR",
+    )
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e251(msg, _node, source_lines=None):
+    """Render a PEP8 unexpected spaces around keyword / parameter equals message."""
+    line = msg.line - 1
+    res = re.search(r"column (\d+)", msg.msg)
+    col = int(res.group().split()[-1])
+    start_idx, end_idx = col, col + 1
+
+    while source_lines[line][end_idx].isspace():
+        end_idx += 1
+
+    if source_lines[line][col - 1] == "=":
+        start_idx -= 1
+    elif source_lines[line][end_idx] == "=":
+        end_idx += 1
+
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (msg.line, slice(start_idx, end_idx), LineType.ERROR, source_lines[line])
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e261(msg, _node, source_lines=None):
+    """Render a PEP8 at least two spaces before inline comment message."""
+    line = msg.line - 1
+    res = re.search(r"column (\d+)", msg.msg)
+    col = int(res.group().split()[-1])
+
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (
+        msg.line,
+        slice(col, len(source_lines[line])),
+        LineType.ERROR,
+        source_lines[line] + "  # INSERT TWO SPACES BEFORE THE '#'",
+    )
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e265(msg, _node, source_lines=None):
+    """Render a PEP8 block comment should start with '# ' message."""
+    line = msg.line - 1
+    yield from render_context(line - 1, line + 1, source_lines)
+    yield (
+        msg.line,
+        slice(0, len(source_lines[line])),
+        LineType.ERROR,
+        source_lines[line] + "  # INSERT SPACE AFTER THE '#'",
+    )
+    yield from render_context(msg.line + 1, msg.line + 3, source_lines)
+
+
+def render_pep8_errors_e266(msg, _node, source_lines=None):
+    """Render a PEP8 too many leading ‘#’ for block comment message."""
+    line = msg.line - 1
+    msg_line_start_index = 0
+
+    while source_lines[line][msg_line_start_index] == " ":
```

Comment:
it's more appropriate for this error to first get the reported column number and then search forward/backward for consecutive `#` to highlight the entire sequence of multiple `#`s